### PR TITLE
keep using the block inversion algorithm in OPM

### PR DESCRIPTION
**Pull request by totto82**
This fixes the issue with dune 2.5 and the polymer cases in opm-data

I have tested this on several of the polymer and co2 cases and plan to run a test on the model 2 realizations.

### CODE REVIEW

**Comment by totto82** [comment]
Should this be part of the release? If so, it needs to be tested thorough.

File: opm/autodiff/ISTLSolver.hpp [hunk 1]
```diff
@@ -479,16 +479,15 @@ namespace Detail
...
+
+	// 3x3 matrix block inversion was unstable at least 2.3 until and including
+	// 2.5.0. There may still be some issue with the 4x4 matrix block inversion
+	// we therefore still use the block inversion in OPM
         typedef ParallelOverlappingILU0<Dune::BCRSMatrix<Dune::MatrixBlock<typename Matrix::field_type,
                                                                            Matrix::block_type::rows,
                                                                            Matrix::block_type::cols> >,
-                                        Vector, Vector> SeqPreconditioner;
-#endif
+                                        				   Vector, Vector> SeqPreconditioner;
```
Comment (atgeirr) [inline opm/autodiff/ISTLSolver.hpp]: There are tabs here, instead of spaces.

**Comment by atgeirr** [comment]
I think perhaps we must also consider the well equation blocks. Will investigate a little.

**Comment by atgeirr** [comment]
> I think perhaps we must also consider the well equation blocks. Will investigate a little.

It looks ok, as the well code uses FMatrixHelper::invertMatrix(), for which a specialized 4x4 overload is provided (in ISTLSolver.hpp).

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
I think this should be part of the release. See also #1465, which fixes a bug that was discovered working on this issue. That PR is not as critical as this, I think. However if we test both together we might save ourselves one round of heavy testing?

**Comment by atgeirr** [comment]
Jenkins is green, and I can confirm that it fixes the problems with the polymer 2D test on my (Dune 2.5-using) system. @totto82 you can merge to master (and pick to release) when satisfied with testing on real cases.

**Comment by totto82** [comment]
The testing is done. I will merge this and update the release branch. I suggest waiting with #1465 to after the release.

**Pull request merged**